facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java

```diff
@@ -962,21 +965,33 @@ private java.util.function.Predicate<BuildConfigurationAudited> getBCAPredicateW
         return predicate;
     }
 
-    private Optional<Build> readLatestRunningBuild(java.util.function.Predicate<BuildTask> predicate)
+    private Optional<Build> readLatestRunningBuild(String query, java.util.function.Predicate<BuildTask> predicate)
             throws RemoteRequestException, MissingDataException {
-        Comparator<BuildTask> comparator = Comparator.comparing(BuildTask::getSubmitTime)
-                .thenComparingLong(task -> parseId(task.getId()).getLongId());
+        java.util.function.Predicate<Build> streamPredicate = (f) -> true;
+        if (!StringUtils.isEmpty(query)) {
+            streamPredicate = rsqlPredicateProducer.getStreamPredicate(query);
+        }
```

Comment:
This makes it "effectively final" (that's java term, probably what you meant by immutable variables). 
```suggestion
        java.util.function.Predicate<Build> streamPredicate;
        if (StringUtils.isEmpty(query)) {
            streamPredicate = (f) -> true;
        } else {
            streamPredicate = rsqlPredicateProducer.getStreamPredicate(query);
        }
```